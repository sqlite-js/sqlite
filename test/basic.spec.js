const { version, Sqlite } = require('..');
const path = require('path');

describe('basic', () => {
  it('should get sqlite version', () => {
    // Ideally we could snapshot test this but we can't do this until
    // we test against a fixed version of SQLite
    expect(version()).toContain('.');
  });

  it('should execute query statements', async () => {
    const conn = new Sqlite();

    await conn.open({
      verbose: true,
      database: path.join(__dirname, 'test.db')
    });

    conn.execute('DROP TABLE IF EXISTS contacts;');

    conn.execute('CREATE TABLE contacts (t VARCHAR PRIMARY KEY);');

    conn.execute(`
      INSERT INTO contacts (t)
      VALUES ("john");
    `);

    expect(
      await conn.execute(`
      SELECT *
      FROM contacts;
    `)
    ).toMatchSnapshot();

    conn.execute('DROP TABLE contacts;');
  });

  it('should format errors', async () => {
    const conn = new Sqlite();

    await conn.open({
      verbose: true,
      database: path.join(__dirname, 'test.db')
    });

    conn.execute('DROP TABLE IF EXISTS contacts;');

    conn.execute('CREATE TABLE contacts (t VARCHAR PRIMARY KEY);');

    expect(() => {
      conn.execute(`
        INSERT INTO contacts (t)
        VALUES (asdfasdf);
      `);
    }).toThrowErrorMatchingSnapshot();
  });
});
