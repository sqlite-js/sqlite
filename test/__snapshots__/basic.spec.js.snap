// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic should execute query statements 1`] = `
Array [
  "john",
]
`;

exports[`basic should format errors 1`] = `"internal error in Neon module: called \`Result::unwrap()\` on an \`Err\` value: SqliteFailure(Error { code: Unknown, extended_code: 1 }, Some(\\"no such column: asdfasdf\\"))"`;
